src/homework5/Calculator.java

```diff
@@ -0,0 +1,25 @@
+package homework5;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+
+public class Calculator {
+
+    Map<Character, BinaryOperator<Double>> map;
+
+    public Calculator() {
+        map = new HashMap<>();
+        map.put('+', (a, b) -> (a + b));
+        map.put('-', (a, b) -> (a - b));
+        map.put('*', (a, b) -> (a * b));
+        map.put('/', (a, b) -> (a / b));
+        map.put('p', (a, b) -> Math.pow(a, b));
+        map.put('@', (a, b) -> Math.pow((a + b) / (a + 117), b));
+    }
+
+    @FunctionalInterface
```

Comment:
Why do you need this interface?  Just implement method int class Calculator with the method 
`double calculate(double x, char operand, double y){....}`.